Mathlib/Algebra/Group/Even/Basic.lean

```diff
@@ -45,6 +45,8 @@ for some `r : α`. -/
 for some `r : α`."]
 def IsSquare (a : α) : Prop := ∃ r, a = r * r
 
+@[to_additive] lemma isSquare_iff (a : α) : IsSquare a ↔ ∃ r, a = r * r := Iff.rfl
```

Comment:
Do we really need this? You can already do `rw [IsSquare`

Reply:
My intention was to future-proof against definition changes and avoid possible defeq abuse in my code. This is can be used in more places eg `rw [IsSquare]` eg `(hx : IsSquare x) ... rcases hx with ...` becomes `rcases (isSquare_iff x).mp hx`.

I'm unaware of the existing conventions around definitions, though.

Reply:
I doubt this definition will change ever again, so I don't think it's worth the trouble of avoiding defeq abuse

Reply:
OK, will revert.
tbh I don't really understand when defeq is "bad" -- do you know of any resources documenting it, or is it just vibes?

Reply:
It's mostly vibes, but I'm sure we could extract some nice guideline from all the examples we have in mathlib